src/layouts/BaseLayout/hooks/useUserAccessGate.js

```diff
@@ -31,47 +33,88 @@ function useOnboardingRedirect({ username }) {
 
 const useUserAccessGate = () => {
   const { provider } = useParams()
-  const { termsOfServicePage, defaultOrgSelectorPage } = useFlags({
-    termsOfServicePage: false,
-    defaultOrgSelectorPage: false,
-  })
-  const { data, isLoading, isSuccess } = useUser({
+  const currentRoute = useRouteMatch()
+
+  const { termsOfServicePage, defaultOrgSelectorPage, sentryLoginProvider } =
+    useFlags({
+      termsOfServicePage: false,
+      defaultOrgSelectorPage: false,
+      sentryLoginProvider: false,
+    })
+
+  const {
+    data: userData,
+    isLoading: userIsLoading,
+    isFetching: userIsFetching,
+    isSuccess: userIsSuccess,
+  } = useUser({
     suspense: false,
-    enabled: !!provider || !config.IS_SELF_HOSTED,
+    enabled: !!provider && !config.IS_SELF_HOSTED,
   })
 
-  useOnboardingRedirect({ username: data?.user.username })
+  const {
+    data: internalUser,
+    isLoading: internalUserIsLoading,
+    isFetching: internalUserIsFetching,
+    isSuccess: internalUserIsSuccess,
+  } = useInternalUser({
+    enabled: !config.IS_SELF_HOSTED,
+    retry: 0,
```

Comment:
Can we do `retry = false` as listed in https://tanstack.com/query/v4/docs/react/guides/query-retries just to make sure it’s set correctly